File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        max_cons_length = 0
        
        for num in nums_set:
            if num-1 not in nums_set:
                curr_length = 1
                while num + 1 in nums_set:
                    curr_length += 1
                    num = num + 1
                max_cons_length = max(max_cons_length, curr_length)
        return max_cons_length

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num : nums) {
            set.add(num);
        }
        
        int maxConsecutiveSeqLength = 0;
        
        for(int num: nums) {
            if(!set.contains(num-1)) {
                int currLength = 1;
                while(set.contains(num + currLength)) {
                    currLength ++;
                }
                maxConsecutiveSeqLength = Math.max(currLength, maxConsecutiveSeqLength);
            }
        }
        return maxConsecutiveSeqLength;
    }
}